Userland/Libraries/LibGfx/ImageFormats/WebPLoader.cpp

```diff
@@ -14,6 +17,34 @@
 
 namespace Gfx {
 
+static ErrorOr<NonnullRefPtr<Bitmap>> rotate_bitmap(NonnullRefPtr<Bitmap> bitmap, Gfx::TIFF::Orientation orientation)
+{
+    if (orientation == Gfx::TIFF::Orientation::Default)
+        return bitmap;
+
+    auto oriented_bmp = TRY(ExifOrientedBitmap::create(orientation, bitmap->size(), bitmap->format()));
+
+    for (int y = 0; y < bitmap->size().height(); ++y) {
+        for (int x = 0; x < bitmap->size().width(); ++x) {
+            auto pixel = bitmap->get_pixel(x, y);
+            oriented_bmp.set_pixel(x, y, pixel.value());
+        }
+    }
+
+    return oriented_bmp.bitmap();
+}
+
+static Gfx::TIFF::Orientation extract_orientation_from_exif(uint8_t const* exif_data, size_t exif_size)
+{
+    auto exif_metadata_result = Gfx::TIFFImageDecoderPlugin::read_exif_metadata({ exif_data, exif_size });
+    if (exif_metadata_result.is_error())
+        return Gfx::TIFF::Orientation::Default;
+
+    auto& exif_metadata = exif_metadata_result.value();
+
+    return exif_metadata->orientation().value_or(Gfx::TIFF::Orientation::Default);
+}
```

Comment:
These methods don't have anything to do specifically with WebP loading, right? I'd rather have some helper methods in `ExifOrientedBitmap`.

Reply:
I moved it to ExifOrientedBitmap

Reply:
`rotate_bitmap` is still WebP-specific; why not centralize that as well?